### ISSUE TITLE
core/build_api.cmake:19 - CMAKE_SYSTEM_NAME is blank

### ISSUE THREAD

**Issue by vladislavmarkov** [forGGe/theCore-blinky#3]
Hello,
I'm using CMake 3.9.6. I got this error while going through instructions listed in README.md.
```
$ nix-shell core --pure
[nix-shell:~/test/theCore-blinky]$ mkdir build
[nix-shell:~/test/theCore-blinky]$ cd build
[nix-shell:~/test/theCore-blinky/build]$ cmake -DCMAKE_TOOLCHAIN_FILE=../core/toolchains/arm-cm4-gnu.cmake ..
CMake Error at core/build_api.cmake:19 (if):
  if given arguments:

    "Linux" "STREQUAL"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:4 (include)


-- Configuring incomplete, errors occurred!
```
CMAKE_SYSTEM_NAME is blank unless "project(whatever)" specified prior to reading CMAKE_SYSTEM_NAME variable.

**Comment by forGGe** [comment]
Hi @vladislavmarkov ! Thanks for reporting. I will look to it ASAP.

**Comment by forGGe** [comment]
Fixed in #4, merged.
Let me know if that works for you. 

This blinky project is working with latest release, but many cool features were implemented since then.  
Check examples in https://forgge.github.io/theCore/examples.html to get idea of what's new there.

**Comment by vladislavmarkov** [comment]
Yup. Just checked this. Now everything is cool. And thanks for fixing the 'template C linker' stuff, btw.
